### ISSUE TITLE
Document `reportDisables` in customizing guide

### ISSUE THREAD

**Issue by henryruhs** [stylelint/stylelint#6584]
### What is the problem you're trying to solve?

I wrote a `no-disable` rule that only works with `"ignoreDisables": true` flag as `root.walkComments` does not contain those declarations that have been disabled. Same issue for disabled files.

```
{
	"plugins":
	[
		"@isnotdefined/stylelint-plugin"
	],
	"rules":
	{
		"@isnotdefined/no-disable": true
	},
	"ignoreDisables": true
}
```

Source: https://github.com/isnotdefinedcom/stylelint-plugin/blob/master/src/rules/no-disable/rule.ts

### What solution would you like to see?

Either there should be an internal API to set `ignoreDisables` once the rule is enabled or an option of a "pure walker" that includes every declaration, even the disabled ones.

**Comment by jeddy3** [comment]
@henryruhs Thanks for the request and for using the template. I saw your `no-obsolete` plugin. It's nice!

The disable comments are an integral part of using Stylelint and shouldn't be disabled. We'll be making [this more explicit](https://github.com/stylelint/stylelint/blob/2e49275d0147a702607feba0b1a96993f607551b/docs/user-guide/customize.md#strictness) in [our upcoming customizing guide](https://github.com/stylelint/stylelint/blob/2e49275d0147a702607feba0b1a96993f607551b/docs/user-guide/customize.md) that'll accompany `stylelint@15.0.0`.

The [original use-case](https://github.com/stylelint/stylelint/issues/1184) for the `--ignore-disables` flag was better addressed by the [`--report-needless-disables`](https://stylelint.io/user-guide/configure#reportneedlessdisables) one. It was a mistake to [also add `--ignore-disables`](https://github.com/stylelint/stylelint/pull/1750), and we should deprecate it at some point. 

Since adding`--report-needless-disables`, we've added more [`report*`](https://stylelint.io/user-guide/configure#report) options and made them customisable. We also added the [`reportDisables`](https://stylelint.io/user-guide/configure#reportdisables) secondary option to the [`rules`](https://stylelint.io/user-guide/configure#rules) configuration property that allows users to report disables _on a rule-by-rule basis_. Users should use these built-in options to manage their disable comments, for example, they could, if they really wanted to (although not recommended), turn on `reportDisables` for every rule to mimic the behaviour of the plugin.

As such, we'll unlikely add `ignoreDisables` to the Node.js API. I'll label as a discussion for a while, though, in case anyone else has thoughts.

**Comment by henryruhs** [comment]
Thank you for this full summary and your kind words.

I played with the `report*` secondary options in my testing suite but neither of them found "disable comments".

```typescript
await lint(
{
	files: './tests/providers/no-disable.css',
	config:
	{
		plugins:
		[
			'./src'
		],
		rules:
		{
			'@isnotdefined/no-disable':
			[
				true,
				{
					reportDisables: true,
					reportNeedlessDisables: true,
					reportInvalidScopeDisables: true,
					reportDescriptionlessDisables: true
				}
			]
		}
		//ignoreDisables: true
	}
});
```

Source: https://github.com/isnotdefinedcom/stylelint-plugin/blob/master/tests/no-disable.spec.ts

In case you plan to `ignoreDisables` it would be nice to provide an alternative, otherwise my plugin will be EOL.

**Comment by jeddy3** [comment]
> Thank you for this full summary

No worries; an explanation only seemed fair.

> I played with the report* secondary options in my testing suite but neither of them found "disable comments".

Only the `reportDisable` option is secondary to the `rules` config property. The other `report*` properties are per config. 

For example, users can disallow disabling your `no-obsolete` rule like so:

```json
{
  "reportNeedlessDisables": true,
  "reportInvalidScopeDisables": true,
  "reportDescriptionlessDisables": true,
  "plugins": ["@isnotdefined/stylelint-plugin"],
  "rules": {
    "@isnotdefined/no-obsolete": [
      true,
      {
        "reportDisables": true
      }
    ]
  }
}
```

Given:

```
/* stylelint-disable @isnotdefined/no-obsolete */
@viewport {}
```

You get:

```console
 1:1  ✖  Rule "@isnotdefined/no-obsolete" may not be disabled     reportDisables                                                                        
 1:1  ✖  Disable for "@isnotdefined/no-obsolete" is missing a description  --report-descriptionless-disables                                
```

> My no-disable rule will be EOL

That makes sense as users can achieve the same (and with more granularity) with the built-in `reportDisable` option

**Comment by henryruhs** [comment]
Now I fully understand it... so my `no-disable` rule is actual build-in to Stylelint if you see it that way?

**Comment by jeddy3** [comment]
> so my no-disable rule is actual build-in to Stylelint?

Yes, but in a granular fashion that aligns with how we recommend people use the disable comments.
